libcst/metadata/name_provider.py

```diff
@@ -17,8 +17,10 @@
     ScopeProvider,
 )
 
+_UNDEFINED_DEFAULT = object
 
-class QualifiedNameProvider(BatchableMetadataProvider[Collection[QualifiedName]]):
+
+class QualifiedNameProvider(BatchableMetadataProvider[_UNDEFINED_DEFAULT]):
```

Comment:
Looks great overall, we just gotta figure out this type here. I'll look at it tomorrow a bit more closely

Reply:
Sure, thanks a lot!